src/common/Makefile.am

```diff
@@ -69,14 +64,32 @@ libcommon_internal_la_SOURCES = \
 	common/addr_parsing.c \
 	common/hobject.cc \
 	common/bloom_filter.cc \
-	common/linux_version.c \
 	common/module.c \
 	common/Readahead.cc \
 	common/Cycles.cc \
 	common/ContextCompletion.cc
 
+if LINUX
+libcommon_internal_la_SOURCES += \
+	common/xattr.c \
+	common/ipaddr.cc \
+	common/ceph_json.cc \
+	common/util.cc \
+	common/pick_address.cc \
+	common/linux_version.c 
+endif
```

Comment:
these are very linux centric. i'll have to confirm each file. do these compile on osx and freebsd for you?